Jsonld_1_1 branch - "intermediate" pull request

danpape opened the pull request:
Please DON'T merge this pull request.

This PR is not ready to merge - it is just being used as an "intermediate" PR to illustrate recent progress on upgrading the JSON-LD API algorithms from the 1.0 to 1.1 specs. It is unfortunate that this branch has been in-progress for several months. Once we get to a good stopping point there will eventually be a "real" PR to get all of these changes merged upstream and any outstanding issues will be fixed after that point.

danpape committed:
update existing expand tests from w3c/json-ld-api

danpape committed:
rename normalize and tdRdf test files

danpape committed:
update existing toRdf tests from w3c/json-ld-api

danpape committed:
rename some existing toRdf tests as done in w3c/json-ld-api

danpape committed:
sort expected toRdf test results before comparing to our results. fixes e029, e060 and e066

danpape committed:
load the expand-manifest.jsonld file and extract TestCases to work on

danpape committed:
working on different kinds of DocumentLoaders

danpape committed:
working on FileLoader

danpape committed:
add missing default operations

danpape committed:
add TestCaseOptions which allow the test manifest to specify test-local options for the library. all current expand tests now running from the manifest

danpape committed:
add latest expand and toRdf test files and manifests

danpape committed:
add new Document types and start getting all tests running from the manifest specs

danpape committed:
add expectErrorCode to Testcase for dealing with NegativeEvaluationTests and refactor a bit

danpape committed:
add all tests in expand manifest as single test functions

danpape committed:
fix expand test t0088

danpape committed:
create test fixture to only load and parse manifest once

danpape committed:
fix expand test te023

danpape committed:
fix expand tests ter02-4

danpape committed:
fix expand test ter06

danpape committed:
fix expand test ter09

danpape committed:
fix expand test ter11

danpape committed:
fix expand tests ter12,ter55

danpape committed:
fix expand test ter14

danpape committed:
remove data file backup dir

danpape committed:
add new test fixture for toRDF tests

danpape committed:
update specVersion for expand test #ter53 and toRdf test #ter53 following update to official test cases

danpape committed:
add new test fixture for normalize tests

danpape committed:
add urdna2015 normalize tests and new code to deal with them, including reading n-quads files directly to an RDFDataset

danpape committed:
updated BNODE regex according to def. at https://www.w3.org/TR/n-quads/\#sec-grammar. fixed normalization urdna15 test 59s parsing issue, but it is now failing for other reasons

danpape committed:
trying to fix expand #t0109; creating some uri parsing tests; we may need to replace uri parsing library

danpape committed:
add uniparser library, v 0.9.4; start replacing usage of the old uri library

danpape committed:
create Uri wrapper for UriParser; use new Uri in JsonLdUrl; fixes expand tests #t0109 and #t0130 and toRDF tests #t0120-#t0132 and #te109

danpape committed:
remove IriUtils class; little cleanup

danpape committed:
enabling expand test #t0077 as we have already implemented expandContext option

danpape committed:
change JsonLdUtils::isAbsoluteIri to use Uri::isAbsolute; detect blank node names a bit better

danpape committed:
first rev of isomorphic rdf dataset comparisons. needs cleanup

danpape committed:
add sha256 algorithm

danpape committed:
upgrade urgna2012 normalization code to urdna2015. fixes most broken urdna2015 tests

danpape committed:
remove old urgna2012 normalization test and test data

danpape committed:
improve utf8 support just enough to fix toRdf tests t0005, te030, te035 and urdna2015 #test060

danpape committed:
improve some test comments

danpape committed:
default processing mode is 1.0 for now

danpape committed:
update uriparser library to v0.9.5

danpape committed:
reenable tests that now work due to uriparser library upgrade

danpape committed:
update embedded rapidcheck code to its latest master branch

danpape committed:
update embedded googletest code to release-1.10.0

danpape committed:
patch googletest to work with recent versions of cmake

danpape committed:
skip any tests marked as specVersion:json-ld-1.1

As we work on upgrading the context processing and term definition
algorithms, we want to just skip most of the tests marked as
specVersion:json-ld-1.1 instead of sliently passing them as we used to
do.

danpape committed:
add comments showing where we need to do more v1.1 code updates; Context::expandValue() is fully updated

danpape committed:
Context::expandIri() is fully updated

danpape committed:
Context::createTermDefinition() and Context::parse() are partially updated

danpape committed:
start work on updating expand()

danpape committed:
add some convenience utility functions suggested in the spec

danpape committed:
fix bug in addValue() after creating a bunch of tests

danpape committed:
fix filename case problem exposed on linux build

danpape committed:
updating context processing and doc expansion to 1.1 spec

110 1.0 tests are passing, 25 are failing.
246 1.1 tests are currently being skipped.

[  FAILED  ] 25 tests, listed below:
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0004
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0015
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0016
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0019
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0023
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0026
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0027
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0029
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0030
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0035
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0036
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0044
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0062
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0063
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0071
[  FAILED  ] JsonLdProcessorExpandTest.expand_t0079
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter02
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter03
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter24
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter32
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter35
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter38
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter39
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter40
[  FAILED  ] JsonLdProcessorExpandTest.expand_ter54

danpape committed:
fixes expand tests 30,35,44,63 and er35

made function arrayContains() because json::array() doesn't really
support .contains() -- it just returns false all the time if the json
var is not an object.

danpape committed:
fixes expand tests 15, 16, 23, 27, 29, 62, 79

need to be careful when using operator[] as json will return newly
created null objects if they aren't already present. using contains() as
a check is helpful.

also, Context::getTermDefinition() also returns a new json::object() if
the term isn't present. this may not be a good idea.

danpape committed:
fixes expand tests 4, 19, 36, and er39

danpape committed:
fixes expand test er38

danpape committed:
"fixes" expand tests 26, 71, er02, er03, er24, and er32

Not really fixes, but simply skips any tests explcitly marked with
specVersion == "json-ld-1.0"

danpape committed:
fixes expand test er40

danpape committed:
fixes expand test er54

danpape committed:
allow all jsonld 1.1 expand tests to run

danpape committed:
fix null ptr dereference; currently running all 1.1 tests, but 177 tests fail

danpape committed:
fix handling of @container entry during term definition; fixes expand tests 80, 94, m17 and m20

danpape committed:
fix handling of @container arrays during term definition; fixes expand tests 82, 84, 85, 87, 96, 98, 99, 101, 105-108

danpape committed:
make some tests for InvalidContainerMapping exclusive of each other; fixes expand tests 83, 86, 97 and 100

danpape committed:
fix handling of @vocab during context processing; fixes expand tests 92 and 124

danpape committed:
better differentiate types of IRI; fixes expand tests 110-112 and 125

danpape committed:
need to check property string values, not property entries; fixes expand tests 114 and pr30

danpape committed:
start find ':' offset from 1 not 0; fixes expand test 117

danpape committed:
don't define terms that match the form of a keyword; fixes expand tests 119 and pr34

danpape committed:
add other missing checks for string that match the form of a keyword; fixes expand tests 120 and pr35-pr39

danpape committed:
if expandedValue is "" then we should save that as null; fixes expand tests 122

danpape committed:
fully update list of valid keywords; fixes expand tests di09, in01-05, in07-09, js01-23, m011-m016

danpape committed:
slightly improve handling of @nest keyword; fixes expand tests en01-en03

hawkmauk committed:
Fix Rdf to RDF in CMakeLists file causing build to fail and remove ; in BlankNode causing compiler warning

hawkmauk committed:
Compile common files for testing into library and wrap API tests in forloop in CMakeLists

danpape committed:
fix extraction of of @import element; fixes expand tests so2,3,5,8,9,11-13

danpape committed:
fix detection of @index within a container element; fixes expand tests pi05-11

danpape committed:
check if @propagate value is boolean; fixes expand test c030

danpape committed:
better handling on default vase direction in context; fixes expand tests di04 and di07

danpape committed:
need to check defined status for value of the entry for value, not just value; fixes expand test er49

danpape committed:
properly append array items; fixes expand test m004

danpape committed:
use map context when needed; fixes expand test m008

danpape committed:
fix handling of language: @none; fixes expand tests m009 and m010

danpape committed:
try to fix scoped context and protected term handling; broke test pr14, but fixes expand tests c001-005, c032-34, c036, pr01, pr03-05, pr08-09, pr11-12, pr15-16, pr19, pr6, pr28, pr31-32, pr40, so07, so10

danpape committed:
need to use pointer for propertyScopedContext to differentiate between non-existent and json::is_null; fixes expand tests pr06 and pr14

danpape committed:
correctly copy active context if needed; fixes expand tests c006-008, c010-012, c019, c020, c022, c025, pr17, pr19, pr20-22, pr25

danpape committed:
correctly copy active context if needed; fixes expand test c014

danpape committed:
check for existence of baseurl before trying to access it; fixes expand test c018

danpape committed:
complete implementation of expand step 4.2; fixes expand test c032 and c024

danpape committed:
properly copy previousContext if needed; clear up logic around @type and @prefix processing; fixes expand tests c009, c013, c015-c017, c021, c027, c028, pr29 and so06

danpape committed:
enable jsonld 1.1 tests for toRDF

danpape committed:
enable jsonld 1.1 tests for toRDF

hawkmauk committed:
snapshot

hawkmauk committed:
Building Prefix from RdfData

hawkmauk committed:
Earl header output correctly

hawkmauk committed:
Moved database to RdfDataBuilder

hawkmauk committed:
Progress on parsing TestResult data

hawkmauk committed:
Merge branch 'danpape:jsonld_1_1' into jsonld_1_1

hawkmauk committed:
Merge branch 'jsonld_1_1' of github.com:runcrypto/jsonld-cpp into test-report

hawkmauk committed:
Creating implementation report

danpape committed:
add libcpr and http-link-header-cpp as external dependency subdirs

danpape committed:
add embedded libcpr version 1.6.2

hawkmauk committed:
Add documentation to configuration Tool

dcdpr reviewed (commented):
is this a comment on a commit, or the pull request?

dcdpr commented on libjsonld-cpp/JsonLdOptions.h:
```diff
@@ -29,7 +29,7 @@ class JsonLdOptions {
      * http://www.w3.org/TR/json-ld-api/#widl-JsonLdOptions-processingMode
      */
-    std::string processingMode_ = JSON_LD_1_0;
+    std::string processingMode_ = JSON_LD_1_1;
```
comment in the file.

dcdpr reviewed (commented):
review on one commit

hawkmauk committed:
Fix typos, remove comments and unimplemented method

hawkmauk committed:
Add checks for nullptr

danpape committed:
Merge pull request #3 from runcrypto/jsonld_1_1

adding ability to generate an implementation report from test results

hawkmauk committed:
Added http-link-header-cpp submodule and updated README.md
Removed jsonld-cpp.doap that is no longer needed
Hotfixed gtest-death-test.cc as per
https://github.com/google/googletest/pull/3024/commits/4679637f1c9d5a0728bdc347a531737fad0b1ca3
to fix compiler error
Explicitly include <limits> in DoubleFormatter.cpp to fix compiler error
Exclude local .vimrc file in .gitignore

fix README link

hawkmauk committed:
Fix issue which only ran one executable
Added default prefix to github profile input on configuration.html
Changes to .gitignore for tempfiles created by development tools

danpape committed:
Merge pull request #4 from runcrypto/jsonld_1_1

danpape committed:
point external/http-link-header-cpp submodule to DCDPR url

danpape committed:
remove unneeded header

danpape committed:
improvements to implementation-report output

danpape committed:
make sure to reset stringstreams that are reused

danpape committed:
fail the test if an JsonLdError is thrown and it's not a NegativeEvaluationTest; 26 toRDF tests are now failing that were previously passing

danpape committed:
get rid of spurious 'empty' test result when running impl report

danpape committed:
clean up some clang compiler warnings

hawkmauk committed:
Tidy up and start docs for implementation report.

hawkmauk committed:
Remove html folder from project directory

hawkmauk committed:
Modified EarlFormatter to use helper functions
added more documentation and regenerated the html

danpape committed:
made few small changes to fix some compiler warnings; also added 'organization' as a person type in the report configuration

hawkmauk committed:
Continuing to add documentation and tidying up implementation report

hawkmauk committed:
Fix merge conflicts

hawkmauk committed:
Fix issue with '.' character in imp-rep description

danpape committed:
make several tweaks to RdfDataBuilder and EarlFormatter and others so our report output more closely matches what is expected

hawkmauk committed:
Documenting RDF data structures

hawkmauk committed:
Merge branch 'jsonld_1_1' of github.com:danpape/jsonld-cpp into jsonld_1_1

hawkmauk committed:
Revamp configuration tool

danpape committed:
add doap:release section to the impl report

hawkmauk committed:
Changes to configuration.js to populate developers when importing JSON
and fix to description when updating state from the form

danpape committed:
Merge branch 'jsonld_1_1' of github.com:danpape/jsonld-cpp into jsonld_1_1

danpape committed:
add report-config.jsonld used to generate first impl report; also set the version number a little lower

danpape committed:
add options for whether to build tests and examples; also make sure to only install jsonld-cpp files

danpape committed:
convert cmake project structure into a superbuild; several other improvements to installation process

This is a combination of 19 commits.

converting cmake project structure into a superbuild

add directory scope to all necessary #include statements

rename libjsonld-cpp directory to jsonld-cpp

more tweaks for superbuild

add directory scope to all necessary #include statements in test code

fix local file loading for example programs

pass CMAKE_BUILD_TYPE and CMAKE_TOOLCHAIN_FILE to subprojects that need them; clean up some warnings from the test code

move tests subdir into jsonld-cpp subdir as it doesn't need to be its own project

be sure to respect CMAKE_INSTALL_PREFIX if set

don't rely on test data file being in a relative location, instead look in a directory defined by env var TEST_RESOURCE_DIR

Update LICENSE

remove some debugging output

update README

delete no longer needed testHelpers file

add top level 'test' target

remove requirement on boost::filesystem

update build instructions

move nlohmann json files out of 'include' and into 'jsonld-cpp' so the build installation makes more sense

danpape committed:
code cleanup

remove nearly unused frameExpansion_ flag from JsonLdOptions

danpape committed:
more cleanup; make sure expansion algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure IRI expansion algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure Value expansion algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure Context Processing algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure Term Definition algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure Generate Blank Node Identifier algorithm is properly commented from the spec

danpape committed:
more cleanup; make sure Node Map Generation algorithm is properly commented from the spec

danpape committed:
more cleanup; redesign of most of the RDF and normalization classes; start simplifying Context class

This is a combination of 51 commits.

wip - create RDFTriple and start converting code to use it

wip - converting going well, 86 toRDF tests are failing, should only be 66 for now

wip: fixed two of the broken toRDF tests

wip: fixed 17 of the broken toRDF tests

wip: fixed last of the broken toRDF tests, plus a bonus 5 more fixed tests

wip: changed what I needed to to get the normalize code running again.

wip: starting to clean up some of the old quad-only based code

wip: more code cleanup, fix some comments, rename functions and classes

wip: more code cleanup, remove unused RDFGraphName class

wip: lots more code cleanup, adding more missing spec comments

wip: recoded most of the normalize code to better match the spec. 18 of 62 tests passing

wip: fix nullptr dereference. 20 of 62 tests passing

wip: replace trying to pass references to just copying a bunch of objects by value. 36 of 62 tests passing

wip: passing CanonicalizationState and BlankNodeNames/issuer by reference (doesn't seem to matter) also adding some debug printing

wip: created RDFTriple2 to replace RDFTriple's use of a json map to store its data

wip: remove old RDFTriple class

wip: rename RDFGraph2 to RDFGraph

wip: rename RDFTriple2 to RDFTriple

wip: extract Node classes to RDFNode files

wip: extract RDFTriple class to RDFTriple files

wip: extract RDFQuad class to RDFQuad files

wip: extract RDFDataset comparison functions to RDFDatasetComparison files

wip: fix and test all constructors and assigment operators for RDFTriple and RDFQuad

wip: fix problem when copy constructing RDFTriple and RDFQuad; fix normalize tests 61 and 62

fix problem in hashNDegreeQuads step 3.1.2; fixes normalize tests 24-29,33-37,40-42,44-46,54,59

wip: debugging urdna2015#test038

wip: fix final four normalize tests, 38,39,47,48

wip: removing lot of debugging printfs

wip: lots of clang warning cleanup

wip: more warning cleanup

wip: remove old normalization code

wip: add ExpansionProcessor class to help break up old JsonLdApi class

wip: add RDFSerializationProcessor class to help break up old JsonLdApi class

wip: add RDFCanonicalizationProcessor class to help break up old JsonLdApi class; remove JsonLdApi class

wip: move functions from RDFCanonicalization class into RDFCanonicalizationProcessor class

wip: simplify RDFDataset api

wip: move toRDF logic out of RDFDataset and into RDFSerializationProcessor

wip: simplify RDFDataset api

wip: more work on RDFDataset api

wip: try to clean up JsonLdOptions and work on expansion api

wip: cleaning up header files

wip: upgrade json.hpp to version 3.11.2; set JSON_USE_IMPLICIT_CONVERSIONS to 0

wip: more header cleanup; also, start to break up parsing logic from Context class

wip: moved Context::expandValue to ExpansionProcessor anon namespace

wip: comment formatting

wip: remove JsonLdConsts::LANGUAGE from contextMap and make it a member variable

wip: remove JsonLdConsts::VOCAB from contextMap and make it a member variable

wip: remove JsonLdConsts::BASE from contextMap and make it a member variable

wip: got rid of contextMap and all functions that referenced it

wip: using simpler version of Context.process

danpape committed:
move some code around for better organization; lot of misc cleanup

This is a combination of 11 commits:

wip: trying to extract logic out of Context and into new class ContextProcessor

wip: more extracting logic out of Context and into new class ContextProcessor

wip: more extracting logic out of Context and into new class ContextProcessor

wip: more extracting logic out of Context and into new class ContextProcessor

wip: working on better commenting, fixing small issues

wip: replace list of keywords to examine with a set

wip: improving some stl usage

wip: fixed references to constants

wip: more cleanup

wip: more cleanup

hawkmauk committed:
CMAKE Add git repository details to ExternalProject http-link-header-cpp

hawkmauk committed:
Resolve linking error during build multiple definition of JsonLdOptions::JSON_LD_1_1

danpape committed:
Merge pull request #9 from runcrypto/improve_installation

Fixes a couple build and linking problems.

danpape committed:
update http-link-header-cpp submodule to latest version

dcdpr commented:
Note this was obsoleted by https://github.com/dcdpr/jsonld-cpp/pull/3
